api/src/main/java/io/opencensus/trace/Span.java

```diff
@@ -158,6 +158,22 @@ public final void addAnnotation(String description) {
   public abstract void addLink(Link link);
 
   /**
+   * Sets the {@link Status} to the {@code Span}.
+   *
+   * <p>If used, this will override the default {@code Span} status. Default is {@link Status#OK}.
+   *
+   * <p>Only the value of the last call will be recorded, and implementations are free to ignore
+   * previous calls. If the status is set via {@link EndSpanOptions.Builder#setStatus(Status)} that
+   * will always be the last call.
+   *
+   * @param status the {@link Status} to set.
+   */
+  public void setStatus(Status status) {
```

Comment:
Should this method also disallow null?

Reply:
We don't do this for any method in Span. I created an issues to add this https://github.com/census-instrumentation/opencensus-java/issues/785